### ISSUE TITLE
feat(www): Language selector dropdown

### ISSUE THREAD

**Issue by tesseralis** [gatsbyjs/gatsby#21746]
## Summary

Implement a language dropdown for gatsbyjs.org.

## Design

Courtesy of @fk 

Figma [clickable prototype](https://www.figma.com/proto/UcsUDRdCoJaQdfITkcKMnU/Language-toggle-UI?node-id=1%3A614&viewport=849%2C4485%2C1&scaling=min-zoom) / [design](https://www.figma.com/file/UcsUDRdCoJaQdfITkcKMnU/Language-toggle-UI?node-id=1%3A41)

![image](https://user-images.githubusercontent.com/21834/65888454-d7b85d80-e39f-11e9-9a99-ef0ca8f8bee7.png)

## Feature list

* "Languages" nav item only appears when `langs` list is non-empty
* Language dropdown appears on mouseover
* Dropdown only shows "featured" languages. For now, these are Spanish (es), Indonesian (id), Japanese (ja), Portuguese Brazilian (pt-BR), Polish (pl), Simplified Chinese (zh-Hans)
* Clicking the link for each language changes the URL to the same page with that locale (e.g. from en/tutorials to ja/tutorials)
* _Bottom link takes users to the `/languages` page_

TODO (@fk): Mobile version of this design

**Comment by fk** [comment]
👋 First thoughts regarding the mobile version: From `360px` device/viewport width up, we don't have any space issues with the current header layout if we drop the "Languages" title/label:

<details>
  <summary>Google Nexus 5X</summary>
  <img width="360" alt="Android - 1" src="https://user-images.githubusercontent.com/21834/75494618-f48fa600-59bc-11ea-94ef-b9adff6e2128.png">
</details>

---

For `320px` keeping the search input like it is now, stuff gets a little crowded. One thing (that is cheap) is to drop the GitHub link, but well, … don't really want to do that. Just to see how that'd look:

<details>
  <summary>iPhone SE</summary>
  <img width="320" alt="iPhone SE - 1" src="https://user-images.githubusercontent.com/21834/75494835-64059580-59bd-11ea-934d-7982af71d4ae.png">
</details>

If we don't want to just "truncate" the search input (and it's placeholder) — which IMO is absolutely fine for the first shot (let's check analytics for those viewport widths). There's other obvious solutions … hello usual suspects:

1. Collapse the search form and just show the icon, like we already do for a range of smaller screens, and e.g. reactjs.org does.
2. Tuck (at least certain) things away in a hamburger menu.

**TL;DR:** _TODO: Look at analytics regarding clicks on the GitHub link and usage stats x devices/viewport dimensions)_

---

Last but not least, the dropdown: Also IMO OK to do that for the first shot, and assuming that we will start out with up to ~5 languages in there. Depending on the number of items this will get icky, esp. if we want to be nice and start to be a little more conscious about landscape orientation.  
Braindumping:

- Link to a basic version of the language switcher page.
- Bottom sheet pattern or similar — sidebar is already there (and could use a rinse, e.g. "tap outside to close" yada yada): [Figma prototype](https://www.figma.com/proto/UcsUDRdCoJaQdfITkcKMnU/Language-toggle-UI?node-id=146%3A382&viewport=110%2C119%2C1&scaling=scale-down)  
  <img width="429" alt="bottom_sheet" src="https://user-images.githubusercontent.com/21834/75498882-ada7ad80-59c8-11ea-823d-9b9a5debcb58.png">

What do y'all think?

(PS: Edited the issue description to contain the link to the Figma "Desktop" design source file along the link to the clickable prototype; hope that was OK?)

**Comment by tesseralis** [comment]
I like it! Thanks for all the detail and research you put into this Flo. I agree that we can start off with the dropdown for v0, and iterate from there. After all, our search bar also doesn't have a specific UI for mobile either :)

**Comment by fk** [comment]
> After all, our search bar also doesn't have a specific UI for mobile either :)

💯 💯 💯 … 🙉 😅

**Comment by id0Sch** [comment]
Hi,
I'd be happy to start working on this.
Just a few questions:
- The current page doesn't look like the mockup, as there are more icons and the search bar is exapnded.
would that be okay?
![image](https://user-images.githubusercontent.com/5864772/77819768-88ed4580-70ee-11ea-98e5-e637b0fdbcfb.png)
- are there any other dropdowns currently in the project, or should I implement one from scratch?


Thanks!
